### PR TITLE
GEOMESA-1758 Transform simple feature wrapper

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
Allows us to read transformed attributes lazily - this can be used when the transform result is going to be discarded and not returned.

File: geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  override def validate() = throw new NotImplementedError
+
+  override def toString = s"TransformSimpleFeature:$getID"
+}
+
+object TransformSimpleFeature {
+
+  import scala.collection.JavaConversions._
+
+  def apply(sft: SimpleFeatureType, transformSchema: SimpleFeatureType, transforms: String): TransformSimpleFeature = {
```
Comment (jnh5y) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala]: I wonder if it'd make sense to expose an easier constructor where one could pass an SFT and property names + transforms and get out the transform schema, transforms, etc.

Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala]: we could, but i think we'd have to move it into index-api - what do you think?

Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala]: the use-case I had was: https://github.com/elahrvivaz/geomesa/blob/6d9229093f23ba53bb1156df5bf03800f6fbb840/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/ArrowBatchIterator.scala#L51-L52

Comment (jnh5y) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala]: Why would it have to move to index-api?

Meh, if I can figure out a way to sort out the call I'm thinking off, I'll add it to this class.

Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/TransformSimpleFeature.scala]: the code to create the transform schema and strings from a query is in index-api

**Pull request merged**